coprocessor/fhevm-engine/zkproof-worker/src/verifier.rs

```
@@ -372,6 +372,12 @@ fn try_verify_and_expand_ciphertext_list(
             keys.pks.parameters(), &keys.public_params,
         ))?;
 
+    // TODO: Make sure we don't try to verify and expand an empty list as it would panic with the current version of tfhe-rs.
+    // Could be removed in the future if tfhe-rs is updated to handle empty lists gracefully.
+    if the_list.is_empty() {
+        return Ok(vec![]);
+    }
```

I think we still need to update the DB to set the outcome to be verified=false - shouldn't we return a specific error we can then use to update?

---

I am considering an empty list as a valid one, IMO it makes everything easier to reason about. But you are right that the ZK proof is not verified in this case (which would be fixed once tfhe-rs is updated). On the other hand, no handle would be created that can be abused.

In general, I think an empty list should be allowed, it's just that we don't verify any proof now.

---

What's the benefit of returning an error?

---

The essential point is that we update the DB so the workers don't get stuck by un-handled proofs that are still selected every time.

---

My reasoning is that we want to eventually support empty input lists, but if that's not the case, I can change as you suggest. Wdyt?

---

We basically need to ensure the query doesn't return this anymore. Either have a separate flag for invalid requests or set verified to false, etc.

---

But the worker itself won't get stuck, it will update verified = true and move on. We never go to the point of the panic this way (expand_and_verify()).

---

Ok, didn't consider it going true would be fine, but you're right!

---

> We basically need to ensure the query doesn't return this anymore. Either have a separate flag for invalid requests or set verified to false, etc.

The query won't return it anymore as everything will succeed.

---

To me the point is whether we want to semantically support empty input lists.

---

That's probably not very relevant since there's no side effects (that I know of at least) to accepting proof on empty lists. Am I missing something?

---

Semantically I'd say any proof is valid for proving "nothing" :)